include/itkMaxPhaseCorrelationOptimizer.hxx

```diff
@@ -163,64 +163,57 @@ MaxPhaseCorrelationOptimizer< TRegistrationMethod >
-        for ( unsigned d = 0; d < ImageDimension; d++ )
-          {
-          dist += ind[d] - oIndex[d];
-          }
-        if ( dist < znSize ) // neighborhood of [0,0,...,0] - in case zero peak is blurred
-          {
-          pixel = oIt.Get();
-          // avoid the initial steep rise of function x/(1+x) by shifting it by 5
-          pixel *= ( dist + 5 ) / ( m_ZeroSuppression + dist + 5 );
-          pixelValid = true;
-          }
-
-        for ( unsigned d = 0; d < ImageDimension; d++ ) // lines/sheets of zero indices
+    constexpr IndexValueType znSize = 4; // zero neighborhood size, in city-block distance
+    mt->ParallelizeImageRegion<ImageDimension>( wholeImage,
+      [&]( const typename ImageType::RegionType& region )
+      {
+        ImageRegionIteratorWithIndex< ImageType > oIt(iAdjusted, region);
+        for (; !oIt.IsAtEnd(); ++oIt)
           {
-          if ( ind[d] == oIndex[d] ) // one of the indices is "zero"
+          bool pixelValid = false;
+          typename ImageType::PixelType pixel;
+          typename ImageType::IndexType ind = oIt.GetIndex();
+          IndexValueType dist = 0;
+          for ( unsigned d = 0; d < ImageDimension; d++ )
             {
-            if ( !pixelValid )
-              {
-              pixel = oIt.Get();
-              pixelValid = true;
-              }
             IndexValueType distD = ind[d] - oIndex[d];
             if ( distD > IndexValueType( size[d] / 2 ) ) // wrap around
               {
               distD = size[d] - distD;
               }
-            double distF = distD * dimFactor[d];
-            // avoid the initial steep rise of x/(1+x) by shifting it by 3% of image size
-            pixel *= ( distF + 3 ) / ( m_ZeroSuppression + distF + 3 );
+            dist += distD;
             }
-          }
 
-        if ( pixelValid ) // either neighborhood or lines/sheets has updated the pixel
-          {
-          oIt.Set( pixel );
+          if ( dist < znSize ) // neighborhood of [0,0,...,0] - in case zero peak is blurred
+            {
+            pixelValid = true;
+            }
+          else
+            {
+            for ( unsigned d = 0; d < ImageDimension; d++ ) // lines/sheets of zero indices
+              {
+              if ( ind[d] == oIndex[d] ) // one of the indices is "zero"
+                {
+                pixelValid = true;
+                }
+              }
+            }
+
+          if ( pixelValid ) // either neighborhood or lines/sheets says update the pixel
+            {
+            pixel = oIt.Get();
+            // avoid the initial steep rise of function x/(1+x) by shifting it by 10
+            pixel *= ( dist + 10 ) / ( m_ZeroSuppression + dist + 10 );
+            oIt.Set( pixel );
+            }
           }
-        }
-    },
-    nullptr );
+      },
+      nullptr );
 
-  WriteDebug( iAdjusted.GetPointer(), "iAdjustedZS.nrrd" );
+    WriteDebug( iAdjusted.GetPointer(), "iAdjustedZS.nrrd" );
```

Comment:
Debug code

Reply:
WriteDebug only writes in debug mode. It does nothing in release mode. See [definition](https://github.com/InsightSoftwareConsortium/ITKMontage/blob/1962de8ab6a00eb698a5a3716a9c6d5d04eb167d/include/itkPhaseCorrelationOptimizer.hxx#L23-L51). We should clean up debugging code as one of the final things. I added an [issue](https://github.com/InsightSoftwareConsortium/ITKMontage/issues/117) to track that.